### ISSUE TITLE
[Simple AA in NewDot] Next Steps for approver are incorrect when Advanced Approval is enabled

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#47933]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**E/E Issue:** https://github.com/Expensify/Expensify/issues/393839
**Issue reported by:** @garrettmknight 
**Slack conversation:** N/A

## Action Performed:
1. Create a Control workspace in OldDot
2. Set an approver that isn't the workspace owner
3. Have an employee that's not the approver or admin submit an expense to the workspace chat in NewDot
4. Open the report as the approver in NewDot

Note: After approving, the "waiting for `admin` to pay" next steps also may be off

## Expected Result:
Next Steps when the report is awaiting the approver's approval should read:
- If you're the approver: Waiting on **You** to approve expense(s).
- If you're not the approver: Waiting on **Approver Name** to approve expense(s).

## Actual Result:
Next steps read:
Waiting on **Workspace Owner** to pay expense(s).

## Workaround:
None

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<img width="1253" alt="Screenshot 2024-08-23 at 11 13 47 AM" src="https://github.com/user-attachments/assets/263f5b19-5633-4ea4-b39b-f4efeeee9c33">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @adelekennedy</details>

**Comment by adelekennedy** [comment]
waiting for an engineer to pick this up

**Comment by aldo-expensify** [comment]
I'm trying to reproduce this, but I get slightly different results... Both, approver and the submitter, see the following:

<img width="528" alt="image" src="https://github.com/user-attachments/assets/103e82b6-274f-4b80-89f6-202386d0308c">

`Waiting for approver@wp-advanced-approval2.com to approve expense(s).`

The approver doesn't see the variation `Waiting on You to approve expense(s).`, so I guess that is something we want to fix?

**Comment by aldo-expensify** [comment]
I see that here we calculate the email we will show in `Waiting for <email> to approve expense(s).`:

https://github.com/Expensify/App/blob/6cb1038e5cbb33ace58f4e27a7bdd2efeb13e8b6/src/libs/NextStepUtils.ts#L63-L74

I don't see any logic there trying to return `You` if you are the approver... so I guess this was really never implemented like that

**Comment by aldo-expensify** [comment]
@garrettmknight is this still reproducible?

**Comment by garrettmknight** [comment]
Looks like this is waiting on the approver now which is good, but I also think it should use the 'you' logic when viewed by the approver:
<img width="403" alt="Screenshot 2024-09-06 at 6 12 21 PM" src="https://github.com/user-attachments/assets/ea11f0de-4299-4619-8217-c4b8571b654e">

**Comment by aldo-expensify** [comment]
> but I also think it should use the 'you' logic when viewed by the approver:

Ok!, I'll investigate using `You` then

**Comment by adelekennedy** [comment]
@aldo-expensify any luck from last week?

**Comment by aldo-expensify** [comment]
No luck last week, but I'm going to work on it now.

**Comment by aldo-expensify** [comment]
I have a draft PR here https://github.com/Expensify/App/pull/49315, I'll make it read for review asap

**Comment by aldo-expensify** [comment]
@garrettmknight I have the following setup:

- Control policy with an approver that is not an admin/owner and an employee. The approver is the default approver (advanced approval is off)
- I submitted a report, which was approved and now it is pending to be paid

From the approver perspective, the next step shows:

<img width="639" alt="image" src="https://github.com/user-attachments/assets/cfb4b05a-77f0-4476-8bd8-ca37663c1e53">


From the policy owner's perspective, the next step shows:

<img width="653" alt="image" src="https://github.com/user-attachments/assets/9c6b51fa-87a4-4930-a6fd-ae6a99507bb2">

This is wrong, right? The policy owner is the one that can pay, but the approver's email appears in the next step

**Comment by aldo-expensify** [comment]
Made the App https://github.com/Expensify/App/pull/49315 ready for review. The PR handles showing `You` in the next step, but it doesn't correct showing the right email when waiting for payment. I'll correct that last part in another PR (I think the bug is in the backend).

**Comment by aldo-expensify** [comment]
The `report.managerID` remains being the approver once the report is approved. Shouldn't have the `report.managerID` to be the policy owner once the report has been approved?

**Comment by aldo-expensify** [comment]
cc @marcochavezf  @Beamanator on https://github.com/Expensify/App/issues/47933#issuecomment-2354201864 and https://github.com/Expensify/App/issues/47933#issuecomment-2354271307 since you may know what is the right behaviour.

**Comment by Beamanator** [comment]
No longer a hot pick since aldo is working on it

**Comment by aldo-expensify** [comment]
> Honestly that's a great question, I'm not tooo too familiar with "who can pay / reimburse" reports 🙃 Happy to bring this to slack to get it clarified if you'd like!

I'll bring it to slack then. There is definitely some inconsistency. The App shows the pay button only for the policy owner, but the "next step message" says that the approver is the one that has to pay.

**Comment by Beamanator** [comment]
Yepppp I feel like admins and owner usually can pay but I’m not 100% sure
when / where

**Comment by aldo-expensify** [comment]
Asking here: https://expensify.slack.com/archives/C06ML6X0W9L/p1726614213444269?thread_ts=1726494273.889699&cid=C06ML6X0W9L

**Comment by aldo-expensify** [comment]
Changing to `Daily`, I'll be working on fixing the "payer" shown in the next step message.

**Comment by aldo-expensify** [comment]
The `payer` account for the next step message is calculated here: https://github.com/Expensify/Web-Expensify/blob/0b0578f71ff37e3324ae7163425da922d53d8f4c/lib/Report/NextStep.php#L856

Under [this setup](https://github.com/Expensify/App/issues/47933#issuecomment-2354201864), we haven't added a bank account, so the "policy's reimburser" is not found, so we fallback to `policyReimburserOrAdminEmail`.

Maybe it is the frontend that needs fixing in this case and we should be showing the "Pay" button for the approver AND the policy owner.

**Comment by aldo-expensify** [comment]
Updating the payer in the next step message seems to have been done here already: https://github.com/Expensify/Web-Expensify/pull/43483

This is the App issue for it https://github.com/Expensify/App/issues/47278.

Considering that, I think there is nothing left to do here besides handling the payment for:

@ishpaul777

**Comment by adelekennedy** [comment]
Payouts due:

- [ ] Contributor+: $250 @ishpaul777 for internal PR review ([Upwork](https://www.upwork.com/nx/wm/offer/104141566))

**Comment by adelekennedy** [comment]
Paid!
